packages/autorest.typescript/CONTRiBUTING.md

```diff
@@ -0,0 +1,348 @@
+# Introduction  
+
+This package contains the High Level Client generation and the Rest Level Client generation from swagger. We will introduce both on how to contribute to high level client generation and rest level client generation.  
+
+## Typescript/Javascript High Level Client SDK generator Design
+
+The generated Typescript/Javascript SDK has the following structure: 
+
+1. Client (Handled by [clientFileGenerator.ts](./src/generators/clientFileGenerator.ts))
+1. Operation Interfaces (Handled by [operationInterfaceGenerator.ts](./src/generators/operationInterfaceGenerator.ts))
+1. Operations (Handled by [operationGenerator.ts](./src/generators/operationGenerator.ts))
+1. Models (Handled by [modelsGenerator.ts](./src/generators/modelsGenerator.ts))
+1. Parameters (Handled by [parametersGenerator.ts](./src/generators/parametersGenerator.ts))
+1. Mappers (Handled by [mappersGenerator.ts](./src/generators/mappersGenerator.ts))
+1. Index File (Handled by [indexGenerator.ts](./src/generators/indexGenerator.ts))
+1. Metadata (Handled [here](./src/generators/static))
+
+The entry point for this design is at [main.ts](./src/main.ts) which invokes the [typescriptGenerator.ts](./src/typescriptGenerator.ts) and starts the entire generation process. You could find all the options that could be sent to the generator at [autorestOptions.ts](./src/utils/autorestOptions.ts)
+
+On a high level, the entire SDK generation process would be:
+
+Swagger Input -> Autorest Core & Modeler -> CodeModel -> TS/JS SDK Generator -> Generated SDK
+
+Within the TS/JS SDK Generator, it has the following stages:
+
+CodeModel + User Options -> Transform CodeModel -> Generate Client -> Generate Client Context -> Generate Operations/Interfaces -> Generate Models -> Generate Parameters -> Generate Index -> Format & Output Generated SDK
+
+### Steps to clone, build & test
+
+1. Use the following command to clone the Typescript/Javascript SDK generator repository:
+
+```
+git clone https://github.com/Azure/autorest.typescript.git
+```
+
+2. Use the following commands to build the SDK generator:
+
+```
+npm install -g @autorest/autorest
+rush update
```

Comment:
I think we also have the rush dependency

```
npm install -g @microsoft/rush
```

Not sure we could talk this in global CONTRIBUTING.md